### PR TITLE
Feat/probabilistic-lgbm-linreg

### PR DESCRIPTION
**Pull request by gnwhr**
First step towards #616.

### Summary
Implements quantile and poisson regression for the `LightGBMModel` and `LinearRegressionModel`.

### CODE REVIEW

**Review by hrzn — COMMENTED** [review]
Looks good! Just got a few comments

File: darts/models/forecasting/gradient_boosted_model.py [hunk 1]
```diff
@@ -48,10 +51,31 @@ def __init__(
...
+        self.quantiles = None
+        self.likelihood = likelihood
+        self._rng = None
+
+        # parse likelihood
+        available_likelihoods = ["quantile", "poisson"]  # to be extended
+        if likelihood is not None:
+            self._check_likelihood(likelihood, available_likelihoods)
+            self.kwargs["objective"] = likelihood
+            self._rng = np.random.default_rng(seed=420)
```
[the remark this answers is not in the document]
Comment (gnwhr) [reply darts/models/forecasting/gradient_boosted_model.py:74]: If I understand correctly, the torch models check and persist the random_state with the `@random_method` decorator. But using this one requires torch. Should we just allow the user to pass  a `seed`?

File: darts/models/forecasting/linear_regression_model.py [hunk 2]
```diff
@@ -44,17 +48,195 @@ def __init__(
...
+                    future_covariates=future_covariates,
+                    max_samples_per_ts=max_samples_per_ts,
+                    **kwargs,
+                )
+
+                self._model_container[quantile] = self.model
+
+            return self
+
+        super().fit(
```
[the remark this answers is not in the document]
Comment (gnwhr) [reply darts/models/forecasting/linear_regression_model.py:173]: does the `return self` statement  in line 171 not exit the method?

File: darts/models/forecasting/regression_model.py [hunk 3]
```diff
@@ -594,3 +596,119 @@ def predict(
...
+            f"If likelihood is specified it must be one of {available_likelihoods}",
+        )
+
+    @staticmethod
+    def _get_model_container():
+        return _QuantileModelContainer()
+
+    def _prepare_quantiles(self, quantiles):
+        if quantiles is None:
+            quantiles = [
```
[the remark this answers is not in the document]
Comment (gnwhr) [reply darts/models/forecasting/regression_model.py:629]: agree

File: darts/models/forecasting/regression_model.py [hunk 4]
```diff
@@ -594,3 +596,119 @@ def predict(
...
+            lam=model_output, size=(*model_output.shape[:2], num_samples)
+        ).astype(float)
+
+    @staticmethod
+    def _ts_like(other: TimeSeries, data: np.ndarray) -> TimeSeries:
+        new_xa = xr.DataArray(data, dims=other._xa.dims, coords=other._xa.coords)
+        return TimeSeries(new_xa)
+
+    @staticmethod
+    def _check_quantiles(quantiles):
```
[the remark this answers is not in the document]
Comment (gnwhr) [reply darts/models/forecasting/regression_model.py:695]: i moved it to `utils/utils.py`

File: darts/models/forecasting/linear_regression_model.py [hunk 5]
```diff
@@ -44,17 +48,195 @@ def __init__(
...
+                        self.model = QuantileRegressor(**self.kwargs)
+                        super().fit(
+                            series=series,
+                            past_covariates=past_covariates,
+                            future_covariates=future_covariates,
+                            max_samples_per_ts=max_samples_per_ts,
+                            **kwargs,
+                        )
+                self.model = QuantileRegressor(**self.kwargs)
+                super().fit(
```
[the remark this answers is not in the document]
Comment (gnwhr) [reply darts/models/forecasting/linear_regression_model.py:161]: changed solver availability check that caused repeated calls to `super().fit()`

File: darts/models/forecasting/linear_regression_model.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hrzn) [reply darts/models/forecasting/linear_regression_model.py:171]: Oh right, maybe you could add an `else` clause, just for clarity here?

**Review by hrzn — APPROVED** [review]
LGTM, thanks!
Got a tiny suggestion to add in `else` clause around line 170 of `linear_regression.py`, then all good.

**Pull request merged**